Mathlib/Data/Polynomial/EraseLead.lean

```diff
@@ -208,6 +229,71 @@ theorem eraseLead_natDegree_le (f : R[X]) : (eraseLead f).natDegree ≤ f.natDeg
   · simp only [h, natDegree_zero, zero_le]
 #align polynomial.erase_lead_nat_degree_le Polynomial.eraseLead_natDegree_le
 
+theorem eraseLead_natDegree_of_nextCoeff (h : f.nextCoeff ≠ 0) :
+    f.natDegree = f.eraseLead.natDegree + 1 := by
+  have hpos := natDegree_pos_of_nextCoeff_ne_zero h
+  suffices f.natDegree - 1 ≤ f.eraseLead.natDegree by
+    have := (add_le_add_iff_right 1).mpr this
+    rw [Nat.sub_add_cancel hpos] at this
+    have : f.natDegree ≥ f.eraseLead.natDegree + 1 := by
+      have := eraseLead_natDegree_le f;
+      have : f.eraseLead.natDegree + 1 ≤ (f.natDegree - 1) + 1 := (add_le_add_iff_right 1).mpr this
+      rwa [Nat.sub_add_cancel hpos] at this
+    linarith
+  have : coeff f (f.natDegree - 1) = coeff f.eraseLead (f.natDegree - 1) := by
+    apply Eq.symm
+    apply eraseLead_coeff_of_ne
+    exact Nat.pred_ne_self (Nat.ne_zero_iff_zero_lt.mpr hpos)
+  rw [nextCoeff, if_neg (natDegree_pos_of_nextCoeff_ne_zero h).ne.symm, this] at h
+  apply le_natDegree_of_ne_zero h
+
+theorem eraseLead_natDegree_of_nextCoeff_eq_zero (h : f.nextCoeff = 0) :
+    f.eraseLead.natDegree ≤ f.natDegree - 2  := by
```

Comment:
```suggestion
    f.eraseLead.natDegree ≤ f.natDegree - 2 := by
```
and check you don't have extraneous spaces anywhere else using a regex like `\S  ` (it will have false positives, but not many)